runner/proxy_stream.go

```diff
@@ -2,45 +2,49 @@ package runner
 
 import (
 	"sync"
+	"sync/atomic"
 )
 
 type ProxyStream struct {
-	sync.Mutex
-	subscribers map[int]*Subscriber
-	count       int
+	mu          sync.RWMutex
```

Comment:
I think you are using wrong, as I know RWMutex is **Allow Multiple Readers**. but it will be block when u try to write.

call RLock, when u  are reading a resource.

Reply:
What do you mean by reading a resource? We don't need the mutex when reading from the reload channel, see here: https://github.com/cosmtrek/air/blob/master/runner/proxy.go#L169. We need the lock when we modify the subscriptions map. So we can create as many subscribers as we want and SSE endpoint will provide updates for all of them in parallel, they will never block each other if that's your concern

Reply:
> We don't need the mutex when reading from the reload channel
so we don't need to use RWMutex, just use mutex is fine. I did see any benefit by using this RWMutex.

the reason is AddSubscriber and RemoveSubscriber both modify the `subscribers` map. 
in my opinion,  RWMutex is going to improve the reading performance.  
can see this answer https://stackoverflow.com/a/19168242